Geometry/HGCalCommonData/plugins/dd4hep/DDHGCalHEFileAlgo.cc

```diff
@@ -0,0 +1,596 @@
+    for (unsigned int i = 0; i < slopeT.size(); ++i)
+      edm::LogVerbatim("HGCalGeom") << "Block [" << i << "] Zmin " << zFrontT[i] << " Rmax " << rMaxFront[i]
+                                    << " Slope " << slopeT[i];
+    edm::LogVerbatim("HGCalGeom") << "DDHGCalHEFileAlgo: NameSpace " << ns.name();
+
+    edm::LogVerbatim("HGCalGeom") << "==>> Constructing DDHGCalHEFileAlgo...";
+    copies.clear();
+#endif
+
+    double zi(zMinBlock);
+    int laymin(0);
+    for (unsigned int i = 0; i < layerNumbers.size(); i++) {
+      double zo = zi + layerThick[i];
+      double routF = HGCalGeomTools::radius(zi, zFrontT, rMaxFront, slopeT);
+      int laymax = laymin + layerNumbers[i];
+      double zz = zi;
+      double thickTot(0);
+      for (int ly = laymin; ly < laymax; ++ly) {
+        int ii = layerType[ly];
+        int copy = copyNumber[ii];
+        double hthick = 0.5 * thickness[ii];
+        double rinB = HGCalGeomTools::radius(zo, zFrontB, rMinFront, slopeB);
+        zz += hthick;
+        thickTot += thickness[ii];
+
+        std::string name = volumeNames[ii] + std::to_string(copy);
+
+#ifdef EDM_ML_DEBUG
+        edm::LogVerbatim("HGCalGeom") << "DDHGCalHEFileAlgo: Layer " << ly << ":" << ii << " Front " << zi << ", "
+                                      << routF << " Back " << zo << ", " << rinB << " superlayer thickness "
+                                      << layerThick[i];
+#endif
+
+        dd4hep::Material matter = ns.material(materials[ii]);
+        dd4hep::Volume glog;
+
+        if (layerSense[ly] < 1) {
+          std::vector<double> pgonZ, pgonRin, pgonRout;
+          if (layerSense[ly] == 0 || absorbMode == 0) {
+            double rmax =
+                (std::min(routF, HGCalGeomTools::radius(zz + hthick, zFrontT, rMaxFront, slopeT)) * cosAlpha) - tol1;
+            pgonZ.emplace_back(-hthick);
+            pgonZ.emplace_back(hthick);
+            pgonRin.emplace_back(rinB);
+            pgonRin.emplace_back(rinB);
+            pgonRout.emplace_back(rmax);
+            pgonRout.emplace_back(rmax);
+          } else {
+            HGCalGeomTools::radius(zz - hthick,
+                                   zz + hthick,
+                                   zFrontB,
+                                   rMinFront,
+                                   slopeB,
+                                   zFrontT,
+                                   rMaxFront,
+                                   slopeT,
+                                   -layerSense[ly],
+                                   pgonZ,
+                                   pgonRin,
+                                   pgonRout);
+            for (unsigned int isec = 0; isec < pgonZ.size(); ++isec) {
+              pgonZ[isec] -= zz;
+              pgonRout[isec] = pgonRout[isec] * cosAlpha - tol1;
+            }
+          }
+
+          dd4hep::Solid solid = dd4hep::Polyhedra(sectors, -alpha, 2. * cms_units::piRadians, pgonZ, pgonRin, pgonRout);
```

Comment:
```suggestion
          dd4hep::Solid solid = dd4hep::Polyhedra(sectors, -alpha, 2._pi, pgonZ, pgonRin, pgonRout);
```

Reply:
We created these user-defined literals in order to avoid multiplication and division.